spec/predicates/human-review-vcs.md

```diff
@@ -0,0 +1,138 @@
+# Predicate type: VCS Code Reviews
+
+Type URI: (tentative) https://in-toto.io/attestation/human-review/vcs/v0.1
+
+Version: 0.1.0
+
+## Purpose
+
+This attestation type is used to describe code review approvals of the simplest
+kind. The general expectation is that reviews are tied to specific changes in a
+version control system (VCS) such as Git.
+
+## Use Cases
+
+Changes to source code are reviewed by other members of the development team.
+The review may be performed using systems like Gerrit or via reviewing
+mechanisms built into version control forges such as GitHub and GitLab.
+
+### Threshold of Reviews on Pull Requests
+
+GitHub allows repository owners to create protection rules that require a
+threshold of approvals before a pull request can be merged. This predicate can
+be used to express meeting such a policy. In this case, GitHub would issue a
+signed attestation identifying all the reviewers for a pull request. The
+subject of the attestation is the underlying commit at the tip of the feature
+branch. This commit, therefore, is one of the parents of the merge commit if it
+is created. If an option that requires rebasing (squash and merge / rebase and
+merge) is selected, the newly created tip commit is used as the attestation
+subject.
+
+## Model
+
+Typically, software development processes incorporate some type of code review.
+Other members of the development team are expected to study the changes
+proposed and approve them, confirming they meet various expectations pertaining
+to code quality, functionality, and so on. Therefore, the general expectation
+is that this attestation is generated as a confirmation, an approval of some
+specific change, and is performed by developers in the supply chain who have
+code review responsibilities.
+
+As this attestation is generated by a version control service or code review
+service, the signatures are from the service rather than the specific users
+issuing the review.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [
+        {
+            "name": "<REPO>",
+            "digest": { "gitCommit": "<HASH>" },
+            "uri": "<LINK TO REPO>"
+        },
+    ],
+    "predicateType": "http://in-toto.io/attestation/human-review/vcs/v0.1",
+    "predicate": {
+        "reviewers": ["<ResourceDescriptor>", ...],
+        "target": "<ResourceDescriptor>",
+        "reviewLink": "<LINK TO REVIEW>",
+        "reviewTime": "<TIMESTAMP>",
+        "annotations": {...}
+    }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the
+[in-toto Attestation Framework's parsing rules](../v1/README.md#parsing-rules).
+
+### Fields
+
+`reviewers` _array of ResourceDescriptor_, _required_
+
+Indicates uniquely the identity of one or more reviewers. MUST NOT be used over
```

Comment:
The source of the attestation is always the pull request product, right?

Reply:
Can you elaborate on `source`? Do you mean the subject? The subject of the attestation identifies the tip of the PR feature branch.